view/frontend/web/js/view/payment/method-renderer/omise-offline-conveniencestore-method.js

```diff
@@ -2,60 +2,60 @@ define(
     [
         'jquery',
         'ko',
+        'Omise_Payment/js/view/payment/omise-base-method-renderer',
         'Magento_Checkout/js/view/payment/default',
         'Magento_Checkout/js/model/full-screen-loader',
         'Magento_Checkout/js/action/redirect-on-success',
         'Magento_Checkout/js/model/quote',
-        'Magento_Customer/js/view/customer',
-        'Magento_Checkout/js/model/error-processor',
-        'Magento_Catalog/js/price-utils',
-        'mage/validation',
+        'Magento_Catalog/js/price-utils'
     ],
     function (
         $,
         ko,
+        Base,
         Component,
         fullScreenLoader,
         redirectOnSuccessAction,
         quote,
-        customer,
-        errorProcessor,
         priceUtils
     ) {
         'use strict';
 
-        const convStoreMinimumPurchaseAmount = 200;
-
-        return Component.extend({
+        return Component.extend(Base).extend({
             defaults: {
                 template: 'Omise_Payment/payment/offline-conveniencestore-form'
             },
 
             isPlaceOrderActionAllowed: ko.observable(quote.billingAddress() != null),
 
-            /**
-             * Get payment method code
-             *
-             * @return {string}
-             */
-            getCode: function () {
-                return 'omise_offline_conveniencestore';
-            },
+            code: 'omise_offline_conveniencestore',
+            restrictedToCurrencies: ['jpy'],
+
 
             /**
-             * Is method available to display
+             * Hook the placeOrder function.
+             * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
              *
              * @return {boolean}
              */
-            isActive: function () {
+            placeOrder: function(data, event) {
+                var failHandler = this.buildFailHandler(this);
+
+                event && event.preventDefault();
+
+                this.getPlaceOrderDeferredObject()
+                    .fail(failHandler)
+                    .done(function() {
+                        redirectOnSuccessAction.execute();
+                    });
+
                 return true;
             },
-
             /**
-             * Get a checkout form data
-             *
-             * @return {Object}
-             */
+            * Get a checkout form data
```

Comment:
Broken indenting on comment header

Reply:
that is broken, or it was broken.. not sure.. this one is result of auto formatting in vc

Reply:
compare to another part of the code, I think VC is tricking us : ) (should have one more space)